src/main/resources/Search/Search.drl

```diff
@@ -0,0 +1,136 @@
+      $a : Availability(judgeId == $judgeId)
+      $s1 : Session(judgeId == $judgeId && DateTimeUtils.contains($a.start, $a.end, start, end))
+      $s2 : Session(judgeId == $judgeId && DateTimeUtils.contains($a.start, $a.end, start, end)
+                     && start > $s1.end)
+      not Session(judgeId == $judgeId && DateTimeUtils.contains($a.start, $a.end, start, end)
+                  && start.isAfter($s1.end)
+                  && start.isBefore($s2.start))
+then
+      insertLogical(new BookableJudge($judgeId, $s1.getEnd(), Duration.between($s1.getEnd(), $s2.getStart())));
+end
+
+/* Room bookable*/
+rule "Room without sessions is always bookable"
+when
+      $j : Room($roomId: id)
+      $a : Availability(roomId == $roomId)
+      not Session(roomId == $roomId && DateTimeUtils.contains($a.start, $a.end, start, end))
+then
+    insertLogical(new BookableRoom($j.getId(), $a.getStart(), $a.getDuration()));
+end
+
+rule "Room is bookable before first session"
+when
+      $j : Room($roomId: id)
+      $a : Availability(roomId == $roomId)
+      $s : Session(roomId == $roomId && DateTimeUtils.contains($a.start, $a.end, start, end))
+      not Session(roomId == $roomId && DateTimeUtils.contains($a.start, $a.end, start, end)
+                  && start.isBefore($s.start))
+then
+      insertLogical(new BookableRoom($roomId, $a.getStart(), Duration.between($a.getStart(), $s.getStart())));
+end
+
+rule "Room is bookable after last session"
+when
+      $j : Room($roomId: id)
+      $a : Availability(roomId == $roomId)
+      $s : Session(roomId == $roomId && DateTimeUtils.contains($a.start, $a.end, start, end))
+      not Session(roomId == $roomId && DateTimeUtils.contains($a.start, $a.end, start, end)
+                  && end.isAfter($s.end))
+then
+      insertLogical(new BookableRoom($roomId, $s.getEnd(), Duration.between($s.getEnd(), $a.getEnd())));
+end
+
+
+rule "Room is bookable between sessions"
+when
+      $j : Room($roomId: id)
+      $a : Availability(roomId == $roomId)
+      $s1 : Session(roomId == $roomId && DateTimeUtils.contains($a.start, $a.end, start, end))
+      $s2 : Session(roomId == $roomId && DateTimeUtils.contains($a.start, $a.end, start, end)
+                     && start > $s1.end)
+      not Session(roomId == $roomId && DateTimeUtils.contains($a.start, $a.end, start, end)
+                  && start.isAfter($s1.end)
+                  && start.isBefore($s2.start))
+then
+      insertLogical(new BookableRoom($roomId, $s1.getEnd(), Duration.between($s1.getEnd(), $s2.getStart())));
+end
+
+/* search */
+query JudgeAndRoomAvailable(
+      String $judgeId,              // can be null
+      String $roomId,               // can be null
+      Duration $duration,           // mandatory
+      OffsetDateTime $start,        // mandatory
+      OffsetDateTime $end           // mandatory
+      )
+
+      $jb : BookableJudge(
+           judgeId == $judgeId || $judgeId == null,
+           duration >= $duration,
```

Comment:
Whats the need of explicit check for duration if in lines 122 and 123 you check for starttime and endtime?

Reply:
possibly it may not be needed